lib/gpio/pin.toit

```diff
@@ -195,17 +195,18 @@ class Pin:
   */
   wait_for value -> none:
     if get == value: return
+    expected_state := value == 1 ? GPIO_STATE_UP_ : GPIO_STATE_DOWN_
+    state_.clear_state expected_state
     gpio_config_interrupt_ num true
     try:
       // Make sure the pin didn't change to the expected value while we
       // were setting up the interrupt.
       if get == value: return
-      expected_state := value == 1 ? GPIO_STATE_UP_ : GPIO_STATE_DOWN_
-      state := state_.wait_for_state expected_state
-      state_.clear_state expected_state
-      return
+
+      state_.wait_for_state expected_state
     finally:
       gpio_config_interrupt_ num false
+      state_.clear_state expected_state
```

Comment:
For some reason this line is necessary.
I had hoped that it would be enough to clear it in line 199, but without this line the test fails.